test/controllers/ReviewDetailsControllerSpec.scala

```diff
@@ -379,7 +383,7 @@ class ReviewDetailsControllerSpec extends PlaySpec with GuiceOneServerPerSuite w
   def businessDetailsWithAuthorisedUser(reviewDetails: ReviewDetails)(test: Future[Result] => Any) = {
     val userId = s"user-${UUID.randomUUID}"
     builders.AuthBuilder.mockAuthorisedUser(userId, mockAuthConnector)
-    when(mockBackLinkCache.fetchAndGetBackLink(ArgumentMatchers.any())(ArgumentMatchers.any())).thenReturn(Future.successful(None))
+    when(mockBackLinkCache.fetchAndGetBackLink(ArgumentMatchers.any())(ArgumentMatchers.any())).thenReturn(Future.successful(Some("/business-customer/business-verification/ATED/businessForm/LTD")))
```

Comment:
Yes the back link will be equal to this - but it feels as though we need to have additional tests in `BusinessVerificationControllerSpec` that specify more precisely the call to `saveBackLink`. Otherwise we are throwing away coverage if we remove the corresponding ATs.

For example updating some of the tests in `BusinessVerificationControllerSpec` like this:
```
      "if non-uk Agent, continue to ATED NRL page" in new Setup {
        when(mockBackLinkCache.saveBackLink(ArgumentMatchers.eq(null), ArgumentMatchers.eq(Some("/business-customer/business-verification/ATED")))(ArgumentMatchers.any())).thenReturn(Future.successful(None))
        continueWithAuthorisedAgentJson(controller, "NUK", FakeRequest().withJsonBody(Json.parse( """{"businessType" : "NUK"}"""))) { result =>
          status(result) must be(SEE_OTHER)
          redirectLocation(result).get must include(s"/business-customer/nrl/ATED")
        }
      }
```

Reply:
Added

Reply:
If it is straightforward then maybe yes. 

You might want to extract the mocked `saveBackLink` calls to a method so that the intent is expressed clearly in the tests. Something like:
```scala
  private def willSaveBackLink(backLink : String) = 
    when(mockBackLinkCache.saveBackLink(
      ArgumentMatchers.eq(null), 
      ArgumentMatchers.eq(Some(backLink)))(ArgumentMatchers.any()))
      .thenReturn(Future.successful(None))
```

Reply:
I have extracted the method and there are now 3 tests with the "willSaveBackLink" method. Let me know if you feel anything else needs to be modified. Thank you